shimmingtoolbox/coils/coil.py

```diff
@@ -165,32 +148,76 @@ def _create_coil_profile(self, dim):
         return sph_coil_profile
 
 
-def convert_to_mp(shim_setting, manufacturers_model_name):
-    """ Converts the ShimSettings tag from the json BIDS sidecar to the scanner units.
-        (i.e. For the Prisma fit DAC --> uT/m, uT/m^2 (1st order, 2nd order))
+def get_scanner_constraints(manufacturers_model_name, order=2):
+    """ Returns the scanner spherical harmonics constraints depending on the manufacturer's model name and required
+        order
 
     Args:
-        shim_setting (list): List of coefficients. Found in the json BIDS sidecar under 'ShimSetting'.
-        manufacturers_model_name (str): Name of the model of the scanner. Found in the json BIDS sidecar under
-                                        ManufacturersModelName'. Supported names: 'Prisma_fit'.
+        manufacturers_model_name (str): Name of the scanner
+        order (int): Maximum order of the shim system
 
     Returns:
-        list: Coefficients with units converted.
+        dict: The constraints including the scanner name, bounds and the maximum sum of currents.
     """
 
     if manufacturers_model_name == "Prisma_fit":
-        # One can use the Siemens commandline AdjValidate tool to get all the values below:
-        max_current_mp = np.array([2300, 2300, 2300, 4959.01, 3551.29, 3503.299, 3551.29, 3487.302])
-        max_current_dcm = np.array([14436, 14265, 14045, 9998, 9998, 9998, 9998, 9998])
+        constraints = {
+            "name": "Prisma_fit",
+            "coef_channel_minmax": [],
+            "coef_sum_max": None
+        }
+        if order >= 0:
+            constraints["coef_channel_minmax"].append([123100100, 123265000])
+        if order >= 1:
+            for _ in range(3):
+                constraints["coef_channel_minmax"].append([-2300, 2300])
+        if order >= 2:
+            constraints["coef_channel_minmax"].append([-4959.01, 4959.01])
+            constraints["coef_channel_minmax"].append([-3551.29, 3551.29])
+            constraints["coef_channel_minmax"].append([-3503.299, 3503.299])
+            constraints["coef_channel_minmax"].append([-3551.29, 3551.29])
+            constraints["coef_channel_minmax"].append([-3487.302, 3487.302])
```

Comment:
Could be 1 line using `.extend([[-4959.01, 4959.01], [-3551.29, 3551.29], ...])`